src/os/bluestore/bluestore_tool.cc

```diff
@@ -831,6 +842,61 @@ int main(int argc, char **argv)
     jf.close_section();
     jf.flush(cout);
   }
+  else if (action == "create-bdev-label") {
+
+    std::vector<std::string> metadata_files = {
+        "bfm_blocks", "bfm_blocks_per_key", "bfm_bytes_per_block", "bfm_size",
+        "bluefs", "ceph_fsid", "ceph_version_when_created", "created_at",
+        "elastic_shared_blobs", "fsid", "keyring", "kv_backend", "magic",
+        "mkfs_done", "osd_key", "ready", "require_osd_release", "type", "whoami"
+    };
+
+    bluestore_bdev_label_t label;
+    std::vector<uint64_t> valid_positions;
+    bool is_multi = false;
+    int64_t epoch = -1;
+
+    int r = BlueStore::read_bdev_label(cct.get(), devs.front(), &label,
+      &valid_positions, &is_multi, &epoch);
+    
+    if (r < 0 && r != -ENOENT) {
+      cerr << "unable to read label for " << devs.front() << ": "
+           << cpp_strerror(r) << std::endl;
+      exit(EXIT_FAILURE);
+    }
+
+    if (r == -ENOENT) {
+      label = bluestore_bdev_label_t();
+      valid_positions = {0};
+      is_multi = false;
+      epoch = 0;
+    }
+
+    for (const auto& file : metadata_files) {
+        std::ifstream infile(path + "/" + file);
+        if (infile) {
+            std::string value((std::istreambuf_iterator<char>(infile)), std::istreambuf_iterator<char>());
+            label.meta[file] = value;
+        } else {
+            cerr << "Warning: unable to read metadata file: " << file << std::endl;
+        }
+    }
+
+    bool wrote_at_least_one = false;
+    for (uint64_t position : valid_positions) {
```

Comment:
If we didn't read any labels "-ENOENT" than `valid_positions` is empty.
I agree we do not want to go into multi, but we should at least write bdev_label at offset 0.